File: algorithm/LeetCode/HappyNumber/HappyNumber.cpp
class Solution {
public:
    bool isHappy(int n) {
        set<int> sample;
        sample.insert(n);
        int s = 0;
        for(;;){
            while(n != 0){
                s += (n % 10) * (n % 10);
                n /= 10;
            }
            if(s == 1)
                return true;
            else{
                if(sample.find(s) != sample.end())
                    return false;
                sample.insert(s);
                n = s;
                s = 0;
            }
        }
    }
};

File: algorithm/LeetCode/HappyNumber/HappyNumber.java
public class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet<Integer>();
        do{
            if(set.contains(n))
                return false;
            set.add(n);
            int tmp = 0;
            while(n != 0){
                tmp += (n%10)*(n%10);
                n/=10;
            }
            n = tmp;
        }while(n!=1);
        return true;
    }
}